src/email.rs

```diff
@@ -106,23 +122,42 @@ or go to https://{domain}/me/pending-invites to manage all of your crate ownersh
                 login,
                 password,
             } => {
-                SmtpTransport::relay(server)?
-                    .credentials(Credentials::new(login.clone(), password.clone()))
-                    .authentication(vec![Mechanism::Plain])
-                    .build()
-                    .send(&email)
-                    .map_err(|_| server_error("Error in sending email"))?;
+                add_custom_metadata("email_backend", "smtp");
+
+                SmtpTransport::relay(server)
+                    .and_then(|transport| {
+                        transport
+                            .credentials(Credentials::new(login.clone(), password.clone()))
+                            .authentication(vec![Mechanism::Plain])
+                            .build()
+                            .send(&email)
+                    })
+                    .map_err(|e| {
+                        add_custom_metadata("email_error", e);
+                        server_error("Failed to send the email")
+                    })?;
+
+                add_custom_metadata("email_id", message_id);
```

Comment:
is there a reason for adding the `email_id` only after the email error was handled? wouldn't it be better to set it as soon as we generated it?

Reply:
There is no use for the `Message-ID` if Mailgun returned an error: in that case, the mail wouldn't have been sent and wouldn't show up in the Mailgun console, so having the ID offers no practical benefit.